app/src/ui/merge-conflicts/merge-conflicts-dialog.tsx

```diff
@@ -229,20 +237,33 @@ export class MergeConflictsDialog extends React.Component<
 
       const tooltip = editorButtonTooltip(this.props.resolvedExternalEditor)
 
+      const onDropdownClick = this.makeDropdownClickHandler(
+        join(this.props.repository.path, path),
```

Comment:
:art: as this returns a function, but is run for every conflicted file in the list on every render, I have a suggestion to save some time rendering the list of conflicted files - avoid invoking `join` until the the handler is actually invoked:

```diff
diff --git a/app/src/ui/merge-conflicts/merge-conflicts-dialog.tsx b/app/src/ui/merge-conflicts/merge-conflicts-dialog.tsx
index b01f7ccd7..759ba0333 100644
--- a/app/src/ui/merge-conflicts/merge-conflicts-dialog.tsx
+++ b/app/src/ui/merge-conflicts/merge-conflicts-dialog.tsx
@@ -238,7 +238,7 @@ export class MergeConflictsDialog extends React.Component<
       const tooltip = editorButtonTooltip(this.props.resolvedExternalEditor)
 
       const onDropdownClick = this.makeDropdownClickHandler(
-        join(this.props.repository.path, path),
+        path,
         this.props.dispatcher
       )
 
@@ -283,10 +283,12 @@ export class MergeConflictsDialog extends React.Component<
   }
 
   private makeDropdownClickHandler = (
-    absoluteFilePath: string,
+    relativePath: string,
     dispatcher: Dispatcher
   ) => {
     return () => {
+      const absoluteFilePath = join(this.props.repository.path, relativePath)
+
       const items: IMenuItem[] = [
         {
           label: OpenWithDefaultProgramLabel,
```

Reply:
Of course, you might have been wanting to avoid `this.props` inside here (TS will ensure it's bound correctly if you do), but `repository` could be another parameter on the function instead.

Reply:
@shiftkey you guessed it! :) I'm up for having repository be another argument, though. will update!

Reply:
a166c65